diff --git a/libobs/obs-service.c b/libobs/obs-service.c
index d44933f5..ff68c20c 100644
--- a/libobs/obs-service.c
+++ b/libobs/obs-service.c
@@ -35,7 +35,7 @@ const char *obs_service_get_display_name(const char *id)
 
 static obs_service_t *obs_service_create_internal(const char *id,
 		const char *name, obs_data_t *settings, obs_data_t *hotkey_data,
-		bool private)
+		bool _private)
 {
 	const struct obs_service_info *info = find_service(id);
 	struct obs_service *service;
@@ -48,7 +48,7 @@ static obs_service_t *obs_service_create_internal(const char *id,
 	service = bzalloc(sizeof(struct obs_service));
 
 	if (!obs_context_data_init(&service->context, OBS_OBJ_TYPE_SERVICE,
-				settings, name, hotkey_data, private)) {
+				settings, name, hotkey_data, _private)) {
 		bfree(service);
 		return NULL;
 	}
@@ -412,3 +412,10 @@ const char *obs_service_get_output_type(const obs_service_t *service)
 		return service->info.get_output_type(service->context.data);
 	return NULL;
 }
+
+EXPORT int obs_customize_set() {
+    if (obs) {
+        obs->video.bypass_graphics = true;
+    }
+    return 0;
+}
