diff --git a/plugins/obs-qsv11/obs-qsv11.c b/plugins/obs-qsv11/obs-qsv11.c
index 2ac18ab2..0206b298 100644
--- a/plugins/obs-qsv11/obs-qsv11.c
+++ b/plugins/obs-qsv11/obs-qsv11.c
@@ -691,18 +691,28 @@ static bool obs_qsv_encode(void *data, struct encoder_frame *frame,
 
 	// FIXME: remove null check from the top of this function
 	// if we actually do expect null frames to complete output.
-	if (frame)
-		ret = qsv_encoder_encode(
-			obsqsv->context,
-			qsvPTS,
-			frame->data[0], frame->data[1], frame->linesize[0],
-			frame->linesize[1],
-			&pBS);
-	else
-		ret = qsv_encoder_encode(
-			obsqsv->context,
-			qsvPTS,
-			NULL, NULL, 0, 0, &pBS);
+    if (frame) {
+        if (packet->keyframe && packet->force_keyframe) {
+            ret = qsv_encoder_force_encode_keyframe(
+                obsqsv->context,
+                qsvPTS,
+                frame->data[0], frame->data[1], frame->linesize[0],
+                frame->linesize[1],
+                &pBS);
+        } else {
+            ret = qsv_encoder_encode(
+                obsqsv->context,
+                qsvPTS,
+                frame->data[0], frame->data[1], frame->linesize[0],
+                frame->linesize[1],
+                &pBS);
+        }
+    } else {
+        ret = qsv_encoder_encode(
+            obsqsv->context,
+            qsvPTS,
+            NULL, NULL, 0, 0, &pBS);
+    }
 
 	if (ret < 0) {
 		warn("encode failed");
