diff --git a/plugins/obs-qsv11/QSV_Encoder.cpp b/plugins/obs-qsv11/QSV_Encoder.cpp
index dd3a9352..7cbcfba6 100644
--- a/plugins/obs-qsv11/QSV_Encoder.cpp
+++ b/plugins/obs-qsv11/QSV_Encoder.cpp
@@ -127,6 +127,25 @@ int qsv_encoder_encode(qsv_t * pContext, uint64_t ts, uint8_t *pDataY,
 		return -1;
 }
 
+int qsv_encoder_force_encode_keyframe(qsv_t * pContext, uint64_t ts, uint8_t *pDataY,
+        uint8_t *pDataUV, uint32_t strideY, uint32_t strideUV,
+        mfxBitstream **pBS)
+{
+    QSV_Encoder_Internal *pEncoder = (QSV_Encoder_Internal *)pContext;
+    mfxStatus sts = MFX_ERR_NONE;
+
+    if (pDataY != NULL && pDataUV != NULL)
+        sts = pEncoder->EncodeKeyFrame(ts, pDataY, pDataUV, strideY, strideUV,
+            pBS);
+
+    if (sts == MFX_ERR_NONE)
+        return 0;
+    else if (sts == MFX_ERR_MORE_DATA)
+        return 1;
+    else
+        return -1;
+}
+
 int qsv_encoder_close(qsv_t *pContext)
 {
 	QSV_Encoder_Internal *pEncoder = (QSV_Encoder_Internal *)pContext;
