diff --git a/plugins/obs-qsv11/QSV_Encoder_Internal.h b/plugins/obs-qsv11/QSV_Encoder_Internal.h
index def736f2..84f308a7 100644
--- a/plugins/obs-qsv11/QSV_Encoder_Internal.h
+++ b/plugins/obs-qsv11/QSV_Encoder_Internal.h
@@ -71,6 +71,9 @@ public:
 	mfxStatus    Encode(uint64_t ts, uint8_t *pDataY, uint8_t *pDataUV,
 			uint32_t strideY, uint32_t strideUV, mfxBitstream
 			**pBS);
+    mfxStatus    EncodeKeyFrame(uint64_t ts, uint8_t *pDataY, uint8_t *pDataUV,
+            uint32_t strideY, uint32_t strideUV, mfxBitstream
+            **pBS);
 	mfxStatus    ClearData();
 	mfxStatus    Reset(qsv_param_t *pParams);
 
